### PR TITLE
Various fixes and refactors

### PR DESCRIPTION
**Pull request by eliotjordan**
- Fixes issue where running Lando generates an incorrect holding_library.rb file.
- Allows all specs to be run with test runners like vim-test.
- Moves LocationProcessorService and IndexFunctions out of marc_to_solr and into the main app where they are used exclusively.
- Refactors LocationProcessorService from a module of class methods into a real class called LocationMapsGeneratorService.
- Extracts indexer build method into a new IndexerService class.
- Adds tests for IndexerService and LocationMapsGeneratorService
- Moves all specs from marc_to_solr into the main app specs.
- Instructs rubocop to exclude the generated translation map files.

### CODE REVIEW

File: config/initializers/traject_initializer.rb [hunk 1]
```diff
@@ -1,4 +1,7 @@
-require File.join(Rails.root, 'marc_to_solr', 'lib', 'location_processor_service')
+if Rails.env.test? || Rails.env.development?
+  LocationMapsGeneratorService.generate_from_templates
```
Comment (hackartisan) [inline config/initializers/traject_initializer.rb:2]: Does this mean we should remove the readme instructions to copy the files?

**Review by hackartisan — CHANGES_REQUESTED** [review]
This is awesome. I think we just need to remove the copy instructions from the readme.

File: config/initializers/traject_initializer.rb (hunk 1 above)
Comment (eliotjordan) [reply config/initializers/traject_initializer.rb:2]: I didn't see anything in the README about that, but if there is a mention somewhere of copying the files, we should remove it.

Comment (hackartisan) [reply config/initializers/traject_initializer.rb:2]: @eliotjordan it was added in https://github.com/pulibrary/marc_liberation/pull/1013. maybe you need to rebase?

Comment (eliotjordan) [reply config/initializers/traject_initializer.rb:2]: I see it now. I was searching for different terms. Pushed a commit to remove that section.

**Review by hackartisan — APPROVED** [review]

**Review by christinach — APPROVED** [review]

**Pull request merged**